src/Microsoft.DotNet.Tools.Compiler.Native/ArgValues.cs

```diff
@@ -56,8 +57,20 @@ public NativeCompileSettings GetNativeCompileSettings()
 
             if (!string.IsNullOrEmpty(IlcPath))
             {
-                config.IlcPath = IlcPath;
-                config.IlcSdkPath = IlcPath;
+                // We want a directory path. If the user gave us the exact path to the executable
```

Comment:
If you look here, you can see NativeCompileSettings already does a check on directory existence in the setter for ILCPath. 

https://github.com/dotnet/cli/blob/master/src/Microsoft.DotNet.Tools.Compiler.Native/NativeCompileSettings.cs#L123

Maintaining this pattern I am proposing that this change be contained in that setter.

As for the purpose of the ArgValues class, it is intended to be a dumb class for collecting argument values passed to `dotnet-compile-native`. The method `ArgValues.GetNativeCompileSettings()` mostly acts as a bridge between ArgValues and NativeCompilationSettings with Transformations/Validations happening in the respective setters.

I agree though that the value of ArgValues's Existence is questionable. I'm not sure what it gives us over the approach you propose. cc @livarcocc Any thoughts here?